dynamixel_driver/src/dynamixel_driver/dynamixel_io.py

```diff
@@ -94,16 +95,25 @@ def __read_response(self, servo_id):
         data = []
 
         try:
-            data.extend(self.ser.read(4))
-            if not data[0:2] == ['\xff', '\xff']: raise Exception('Wrong packet prefix %s' % data[0:2])
-            data.extend(self.ser.read(ord(data[3])))
-            data = array('B', ''.join(data)).tolist() # [int(b2a_hex(byte), 16) for byte in data]
-        except Exception, e:
+            # read the header to check how many more bytes to read
+            buf = self.ser.read(4)
+            data += struct.unpack('<4B', buf)
+
+            # throw an exception if we have the wrong number of bytes
+            if data[:2] != [0xff, 0xff]:
+                raise Exception('Wrong packet prefix %s' % data[:2])
+
+            # now read the rest of the packet
+            n = data[3]
+            buf = self.ser.read(n)
+            data += struct.unpack('<{}B'.format(n), buf)
+        except Exception as e:
```

Comment:
Deserialization was written in a wonky way -- this uses the `struct` library to be agnostic of python 2/3 behavior. We read bytes off the wire as a list of integers (unsigned bytes), instead of dealing with the actual str/byte behavior before.